File: ReformaAgraria/Models/ToraSubject.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ReformaAgraria.Models
{
    public class ToraSubject: BaseEntity<int>
    {
        public ToraSubject() { }
        
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public override int Id { get; set; }

        public string Name { get; set; }
       
        public MaritalStatus MaritalStatus { get; set; }

        public string Address { get; set; }

        public Gender Gender { get; set; }

        public int? Age { get; set; }

        public EducationalAttainment EducationalAttainment { get; set; }

        public int TotalFamilyMembers { get; set; }

        public string LandStatus { get; set; }

        public string LandLocation { get; set; }

        public decimal Size { get; set; }

        public string PlantTypes { get; set; }

        public string Notes { get; set; }

        public int FkToraObjectId { get; set; }

        [ForeignKey("FkToraObjectId")]
        public ToraObject ToraObject { get; set; }
    }
}

File: ReformaAgraria/ClientApp/app/models/ToraSubject.ts
import { MaritalStatus } from './MaritalStatus';
import { Gender } from './Gender';
import { EducationalAttainment } from './EducationalAttainment';
import { ToraObject } from './ToraObject';
import { BaseEntity } from './BaseEntity';

export interface ToraSubject extends BaseEntity<number> { 
	id: number;
	name: string;
	maritalStatus: MaritalStatus;
	address: string;
	gender: Gender;
	age: number;
	educationalAttainment: EducationalAttainment;
	totalFamilyMembers: number;
	landStatus: string;
	landLocation: string;
	size: number;
	plantTypes: string;
	notes: string;
	fkToraObjectId: number;
	toraObject: ToraObject;
}
